impl/src/main/java/org/jboss/cdi/tck/tests/extensions/configurators/annotatedTypeConfigurator/ProcessAnnotatedTypeObserver.java

```diff
@@ -176,4 +178,35 @@ void observesCountrysidePAT(@Observes ProcessAnnotatedType<Countryside> event) {
         return originalCatAT;
     }
 
+    public boolean annotatedTypesEqual(AnnotatedType<?> firstAT, AnnotatedType<?> secondAT) {
+        return firstAT.getJavaClass().equals(secondAT.getJavaClass()) && firstAT.getBaseType().equals(secondAT.getBaseType()) && firstAT.getAnnotations()
+                .equals(secondAT.getAnnotations()) && firstAT.getTypeClosure().equals(secondAT.getTypeClosure()) && firstAT.getConstructors()
+                .equals(secondAT.getConstructors()) && firstAT.getMethods().equals(secondAT.getMethods()) && firstAT.getFields()
+                .equals(secondAT.getFields());
+
+    }
+
+    public boolean annotatedCallablesEqual(AnnotatedCallable<?> firstAM, AnnotatedCallable<?> secondAM) {
+        return firstAM.getJavaMember().equals(secondAM.getJavaMember()) && firstAM.getDeclaringType().equals(secondAM.getDeclaringType()) && firstAM
+                .getBaseType().equals(secondAM.getBaseType()) && firstAM.getAnnotations().equals(secondAM.getAnnotations()) && firstAM.getTypeClosure()
+                .equals(secondAM.getTypeClosure()) && firstAM.getParameters().equals(secondAM.getParameters());
+
+    }
+
+    public boolean annotatedParametersEqual(AnnotatedParameter<?> firstAP, AnnotatedParameter<?> secondAP) {
+        return firstAP.getJavaParameter().equals(secondAP.getJavaParameter()) && annotatedCallablesEqual(firstAP.getDeclaringCallable(),
+                secondAP.getDeclaringCallable()) && firstAP
+                .getBaseType().equals(secondAP.getBaseType()) && firstAP.getAnnotations().equals(secondAP.getAnnotations()) && firstAP.getTypeClosure()
```

Comment:
Comparing `Type` is likely questionable because it doesn't have to implement `equals` as well???